articles/code-management/about-code-management-csharp/about-code-management-csharp.md

```diff
@@ -0,0 +1,98 @@
+---
+uid: code-management.about-code-management-csharp
+---
+# About Code Management (C#)
+
+This article provides information about [Code Management](xref:code-management.about-code-management) done in a C# context using the `Intent.OutputManager.RoslynWeaver` Module.
+
+## Introduction
+
+In C# files it is quite common to notice this line of code near the top of the file:
+
+```csharp
+[assembly: DefaultIntentManaged(Mode.Fully)]
+```
+
+This attribute is located on a global scope which indicates that it will instruct Intent Architect to treat the whole file according to the configuration mode of `Fully` (in this case).
+
+`Mode` comes with three configuration options:
+
+* `Fully` - It has **full** control over the file for Code Automation (like typical Code Automation).
+* `Merge` - It may only make modifications and additions to this file but it's not allowed to remove anything.
+* `Ignore` - It has **no** control over the file and will not generate or overwrite anything in it.
+
+> [!NOTE]
+> Comments that are made by the developer (not the Code Automation) outside the scope of a `namespace` will not be modified in any way even if `Mode` is set to `Fully`.
+
+## Override different parts using `[IntentManaged]`
+
+Within a C# source file, one may choose to override certain behaviour of how Intent Architect performs code automation by inserting a `[IntentManaged]` attribute right above a class, property, field or method (or various other language elements).
+
+For example, adding this attribute above a method will look something like this:
+
+```csharp
+[IntentManaged(Mode.Ignore)]
+public void ChangeCountry(string country)
+{
+    throw new NotImplementedException();
+}
+```
+
+It will instruct Intent Architect not to modify (or remove) this method in any way during the Software Factory Execution.
```

Comment:
Rather:

> When the RoslynWeaver sees this, it will know not to modify the method in any way during code merging.

Reply:
Cool, I think that's a good idea, we also don't have to have the braces around "or remove", but I don't feel strongly either way.